pkg/compiler/vars.go

```diff
@@ -70,12 +73,17 @@ func (c *varScope) newVariable(t varType, name string) int {
 	case varLocal:
 		return c.newLocal(name)
 	case varArgument:
-		_, ok := c.arguments[name]
+		n = c.arguments.size
+		c.arguments.size++
+		if name == "_" {
+			break
+		}
+
+		_, ok := c.arguments.values[name]
 		if ok {
 			panic("argument is already allocated")
 		}
-		n = len(c.arguments)
-		c.arguments[name] = n
+		c.arguments.values[name] = n
```

Comment:
The other possibility is to use an impossible fake name here like `%%%` (of length `len(c.arguments)`) and not use `size` at all. There shouldn't often be many (>10) arguments and if there are we'll have a (relatively) big map anyway.